File: src/main/java/com/procurement/clarification/controller/PeriodController.java
package com.procurement.clarification.controller;

import com.procurement.clarification.model.dto.bpe.ResponseDto;
import com.procurement.clarification.model.dto.params.PeriodParams;
import com.procurement.clarification.service.PeriodService;
import java.time.LocalDateTime;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "/period")
public class PeriodController {

    private final PeriodService periodService;

    public PeriodController(final PeriodService periodService) {
        this.periodService = periodService;
    }

    @PostMapping("/save")
    public ResponseEntity<ResponseDto> savePeriod(@RequestParam("identifier") final String cpId,
                                                  @RequestParam("country") final String country,
                                                  @RequestParam("stage") final String stage,
                                                  @RequestParam("owner") final String owner,
                                                  @RequestParam("pmd") final String pmd,
                                                  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                  @RequestParam("startDate") final LocalDateTime startDate,
                                                  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                  @RequestParam("endDate") final LocalDateTime endDate) {
        final PeriodParams params = new PeriodParams(cpId, stage, owner, country, pmd, startDate, endDate);
        return new ResponseEntity<>(periodService.calculateAndSavePeriod(params), HttpStatus.CREATED);
    }
}

File: src/main/kotlin/com/procurement/clarification/controller/PeriodController.kt
package com.procurement.clarification.controller

import com.procurement.clarification.model.dto.bpe.ResponseDto
import com.procurement.clarification.model.dto.params.PeriodParams
import com.procurement.clarification.service.PeriodService
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime

@RestController
@RequestMapping(path = ["/period"])
class PeriodController(private val periodService: PeriodService) {

    @PostMapping("/save")
    fun savePeriod(@RequestParam("cpid") cpId: String,
                   @RequestParam("country") country: String,
                   @RequestParam("stage") stage: String,
                   @RequestParam("owner") owner: String,
                   @RequestParam("pmd") pmd: String,
                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                   @RequestParam("startDate") startDate: LocalDateTime,
                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                   @RequestParam("endDate") endDate: LocalDateTime,
                   @RequestParam("setExtendedPeriod") setExtendedPeriod: Boolean): ResponseEntity<ResponseDto> {
        val params = PeriodParams(
                cpId = cpId,
                stage = stage,
                owner = owner,
                country = country,
                pmd = pmd,
                startDate = startDate,
                endDate = endDate,
                setExtendedPeriod = setExtendedPeriod)
        return ResponseEntity(
                periodService.calculateAndSavePeriod(params),
                HttpStatus.CREATED)
    }

    @GetMapping
    fun getPeriod(@RequestParam("cpid") cpId: String,
                  @RequestParam("stage") stage: String): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.getPeriod(cpId = cpId, stage = stage),
                HttpStatus.CREATED)
    }
}
